Clarify the inheritance of the properties at different levels (#1299)
radiantearth/stac-spec: item-spec/item-spec.md

As detailed above, Items contain properties, which are the main source of metadata for searching across Items. Many content
extensions can add further property fields as well. Any property that can be specified for an Item can also be specified for
a specific asset. This can be used to override a property defined in the Item, or to specify fields for which there is no

> single value for all assets. If a property is defined at the STAC entity level (Item or Collection), it is the default value for all assets.
> The values in the top level properties are considered nominal for the the Item or the Collection and thus may be adjusted for
> specific assets (see example below).

Reviewer: Trying to make it more clear, also this is just the Item Spec so Collections should have this paragraph, too.

```suggestion
single value for all assets.
If a property is defined in the Item Properties, it is the default value for all assets.
Assets may override the properties in the Item Properties for specific assets (see example below).
Note that in some cases the inheritance may not lead to the expected results if other semantics have been defined for the property.
See the examples for `gsd` and `eo:bands` below, but it also applies to some other fields such as `created` or `updated`.
```

extensions can add further property fields as well. Any property that can be specified for an Item can also be specified for
a specific asset. This can be used to override a property defined in the Item, or to specify fields for which there is no
single value for all assets.
If a property is defined in the Item Properties, it is the default value for all assets in the Item.

> If a property is defined in a Collection on the top-level, it is the default value for all assets in the Collection.

Reviewer 2: > it is the default value for all assets in the Collection

Not items? E.g. if a `Collection` has `proj:epsg=32613`, does that projection information apply to the item (e.g. in the case of a `proj:geometry` on the item).

Reviewer: No, we only inherit in the same file.

Assets may override the properties inherited for specific assets (see example below).
Note that in some cases the inheritance may not lead to the expected results if other semantics have been defined for the property.
See the examples for `gsd` and `eo:bands` below.
It also applies to some other fields such as `created` or `updated`

> (metadata-related timestamps get inherited and data-related timestamps are defined in assets).

Reviewer: ```suggestion
(metadata-related timestamps would get inherited and mixed with data-related timestamps that are defined in assets).
```

> (metadata-related timestamps would get inherited and mixed with data-related timestamps that are defined in assets).

Reviewer 2: This parenthetical note is confusing to me - I imagine it will be for other people as well.

```suggestion
(inheritance may be undesirable in these cases, e.g. the `created` attribute of an item would describe when the metadata was generated, whereas on an asset the `created` attribute would describe when the data file itself was created).
```

> It also applies to some other fields such as `created` or `updated`
> (inheritance may be undesirable in these cases, e.g. the `created` attribute of an item would describe when the metadata was generated, whereas on an asset the `created` attribute would describe when the data file itself was created).

Reviewer: ```suggestion
It also applies to some other fields such as `created` or `updated`.
Inheritance may be undesirable in these cases, e.g. the `created` attribute of an item would describe when the metadata
was generated, whereas on an asset the `created` attribute would describe when the data file itself was created.
```

extensions can add further property fields as well. Any property that can be specified for an Item can also be specified for
a specific asset. This can be used to override a property defined in the Item, or to specify fields for which there is no
single value for all assets.
If a property is defined in the Item Properties, it is the default value for all assets in the Item.
If a property is defined in a Collection on the top-level, it is the default value for all assets in the Collection.

> Assets may override the properties inherited for specific assets (see example below).

Reviewer: ```suggestion
Assets may override the properties inherited for specific assets (see example below).
Inheritance does not apply across multiple files, i.e. is restricted to a single Item or Collection.
```
